### ISSUE TITLE
Markdown line ending problems in 1.8.15 and master

### ISSUE THREAD

**Issue by tttapa** [doxygen/doxygen#6800]
Version 1.8.15 and the master version seem to replace Markdown line breaks (two spaces at the end of a line) by new paragraphs.  
The master version has the same problem, and it also removes Markdown headers.

Consider the following comment:
```cpp
/**
 * @brief   Test
 * 
 * ### Title
 * Paragraph 1
 * 
 * Paragraph 2··
 * Newline?
 */
```
(Where `··` is a double space at the end of the line.)

In version 1.8.14, this produces the following HTML:
```html
<p>Test. </p>
<h3>Title</h3>
<p>Paragraph 1</p>
<p>Paragraph 2 <br>
Newline?</p>
```
This is correct, in my eyes.

In version 1.8.15, it produces the following HTML:
```html
<p>Test. </p>
<h3>Title</h3>
<p>Paragraph 1</p>
<p>Paragraph 2</p>
<p>Newline?</p>
```
Now the line break is a new paragraph. This means that all previously well-formatted documentation entries are now just mostly whitespace between lines.

In the master version (6009d38), it produces the following HTML:
```html
<p>Test. </p>
<p>Paragraph 1</p>
<p>Paragraph 2</p>
<p>Newline?</p>
```
As you can see, the line break is still wrong, and now it even removes the headings.

It seems to be a general MarkDown problem, as it occurs in function comments, example comments, and `.md` pages.

Thank you.

**Comment by albert-github** [comment]
When I run the example with version 1.8.14 (and 1.8.15 and mentioned master version) I get:
`warning: found subsubsection command outside of subsection context!`
Furthermore I don't see the word "Title" anywhere in HTML output (default doxygen configuration file).

It is quite strange to have a level 3 section here.

Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem? Maybe you can also add the HTML output you get for the 1.8.14, 1.8.15 and master version.

**Comment by tttapa** [comment]
Thank you for your quick response.

I attached a zip with a minimal example to reproduce it (just a single source file with the documentation an empty class, and a doxyfile of two lines).

I exported the HTML of the three different versions, as well as the output from doxygen.  
It does indeed show the warning, but only in the master version.

[Doxygen-Markdown-Newline-Issue.zip](https://github.com/doxygen/doxygen/files/2802619/Doxygen-Markdown-Newline-Issue.zip)

**Comment by albert-github** [comment]
I looked at the "subsubsection" problem and I think that in the past this message was incorrectly not given. 

You can suppress this message and getting the "Title" back, in this case, by setting:
  `TOC_INCLUDE_HEADINGS=0`
in your doxygen configuration file.
It is now consistent with the situation when a `###` header is given in page (i.e. not as part of the comment for a class / function etc.). To get a similar effect your output you can use `<h3>`.

**Comment by tttapa** [comment]
Are headers usually not allowed in the detailed description? 

Any update on the line endings?

**Comment by albert-github** [comment]
Headers are possible but, in my opinion, not really common in detailed descriptions.

Still looking into the double space problem.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6801

**Comment by tttapa** [comment]
It seems to work just fine with #6801 and `TOC_INCLUDE_HEADINGS=0`, thanks a lot!

Is the `TOC_INCLUDE_HEADINGS=0` is just a hack to circumvent a bug, or do I always need it to get headers in my documentation?

**Comment by albert-github** [comment]
I thin you will always need the `TOC_INCLUDE_HEADINGS=0` when you don't have all levels of the sections / subsections etc. included.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
